packages/react-examples/src/react/DetailsList/DetailsList.CumulativeGroupHeaders.Example.tsx

```diff
@@ -0,0 +1,179 @@
+      { name: 'Item 2', count: 2 },
+      { name: 'Item 3', count: 3 },
+      { name: 'Item 4', count: 4 },
+      { name: 'Item 5', count: 5 },
+      { name: 'Item 6', count: 6 },
+    ];
+    this._groups = [
+      {
+        name: 'First Group',
+        key: 'first-group',
+        startIndex: 0,
+        count: 3,
+      },
+      {
+        name: 'Second Group',
+        key: 'second-group',
+        startIndex: 3,
+        count: 3,
+      },
+    ];
+    this._columns = buildColumns(this._items);
+    this.state = { selectedItem: dropdownOptions[0] };
+  }
+
+  public render(): JSX.Element {
+    return (
+      <>
+        <Dropdown
+          label="Checkbox visibility"
+          selectedKey={this.state.selectedItem.key}
+          onChange={this._onChange}
+          options={dropdownOptions}
+          styles={dropdownStyles}
+        />
+        <DetailsList
+          items={this._items}
+          groups={this._groups}
+          groupProps={{
+            onRenderHeader: this._onRenderGroupHeader,
+            headerProps: {
+              onRenderTitle: this._onRenderGroupHeaderTitle,
+            },
+          }}
+          getGroupHeight={this._getGroupHeight}
+          ariaLabelForSelectionColumn="Toggle selection"
+          ariaLabelForSelectAllCheckbox="Toggle selection for all items"
+          checkButtonAriaLabel="select row"
+          onRenderDetailsHeader={this._onRenderDetailsHeader}
+          checkboxVisibility={CheckboxVisibility[this.state.selectedItem.key as keyof typeof CheckboxVisibility]}
+        />
+      </>
+    );
+  }
+
+  private _onChange = (event: React.FormEvent<HTMLDivElement>, item: IDropdownOption) => {
+    this.setState({ selectedItem: item });
+  };
+
+  private _onRenderDetailsHeader: IDetailsListProps['onRenderDetailsHeader'] = props => {
+    if (props) {
+      return <DetailsHeader {...props} ariaLabelForToggleAllGroupsButton={'Toggle selection'} />;
+    }
+    return null;
+  };
+
+  private _onRenderGroupHeader: IDetailsGroupRenderProps['onRenderHeader'] = props => {
+    if (props) {
+      // This retrieves the measured column width
+      this._columns = props.columns!;
+      return <GroupHeader {...props} />;
+    }
+
+    return null;
+  };
+
+  private _getSumFromGroup = (group: IGroup): number => {
+    const { startIndex, count } = group;
+
+    let sum = 0;
+    for (let i = startIndex; i < startIndex + count; i++) {
+      sum += this._items[i].count;
+    }
+
+    return sum;
+  };
+
+  private _onRenderGroupHeaderTitle: IGroupHeaderProps['onRenderTitle'] = props => {
+    if (props) {
+      const item = {
+        ...props.group,
+        count: this._getSumFromGroup(props.group!),
+      };
+
+      console.log('props', props);
+
+      return (
+        <DetailsRow
+          {...props}
+          styles={detailsRowStyles}
+          groupNestingDepth={props.groupLevel! - 1}
+          columns={this._columns}
+          item={item}
+          itemIndex={-1}
```

Comment:
This also feels hacky but it keeps the header rows from being selected.

🤔 When I started my intent with this PR was to be able to create an example because I believe DetailsList has all the features to support the request in the linked issue however, in my first attempt at writing a robust example I'm left feeling this is a bit on the hacky side for our documentation.

Might need a special "GroupedDetailsHeaderRow" for this situation.